src/main.rs

```diff
@@ -181,8 +181,21 @@ fn run() -> Result<bool> {
 
     match app.matches.subcommand() {
         ("cache", Some(cache_matches)) => {
-            run_cache_subcommand(cache_matches)?;
-            Ok(true)
+            // If there is a file named 'cache' in the current working directory,
+            // arguments for subcommand 'cache' are not mandatory.
+            // If there are non-zero arguments, execute the subcommand cache, else, open the file cache.
+            if !cache_matches.args.is_empty() {
+                run_cache_subcommand(cache_matches)?;
+                Ok(true)
+            } else {
+                let mut config = app.config()?;
```

Comment:
The code duplication in this `else`-clause is kind of unfortunate (it mirrors most steps that we normally do, if no subcommand is given).

I was wondering whether we could simply take out the `cache` subcommand completely if a `cache` file exists. We wouldn't get the behavior which you implemented here (i.e. that `bat cache --init` still works)

Reply:
Umm, yeah I thought about it, like implementing that in the app.rs file, but, umm, how about we make a function for calling the controller and call it first here and then in the other match clause?

Reply:
Or else we can do what you are suggesting, but yep that is gonna drop the functionality there :sweat_smile:

Reply:
Ok, sounds good. We should probably move the `HighlightingAssets::new()` part into this new function (and remove the `assets` argument from `list_languages` and `list_themes` and just call `HighlightingAssets::new()` inside those functions).

Reply:
I will be doing what you just said! Should be ready in a few minutes!